pkg/kudoctl/resources/dependencies/resolve_test.go

```diff
@@ -22,7 +23,9 @@ func (resolver nameResolver) Resolve(
 	appVersion string,
 	operatorVersion string) (*packages.Package, error) {
 	for _, pkg := range resolver.Pkgs {
-		if pkg.Resources.Operator.Name == name {
+		if pkg.Resources.Operator.Name == name &&
+			(operatorVersion == "" || pkg.Resources.OperatorVersion.Spec.Version == operatorVersion) &&
```

Comment:
so many dots here that I would almost create a function to get version on pkg level :D